rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,401 @@
+processing is outside the scope of this RFC.
+
+### Data Portal APIs
+
+The DP Browser App will use these endpoints to interact with the Backend using these APIs. This section describes the new
+endpoints and how they are used. These are the new API endpoints needed to complete an upload.
+
+These APIs replace the [PUT /v1/submission/{project_uuid}/file](https://docs.google.com/document/d/1d8tv2Ub5b3E7Il85adOAUcG8P05N6UBZJ3XbhJSRrFs/edit#heading=h.3hln6w2kzoyt)
+section.
+
+#### PUT submission/{submission_id}/upload/link
+
+An authenticated user can upload a file from a shared link to a data set in their submission. 
+
+If the upload is in an error state, this endpoint can be used with the dataset_id to restart the submission. If a new 
+link is provided, the new link will be used. If the submissions is not in an error state, or is complete, this endpoint 
+will return the current status of the submission.
+
+**Request:**
+
+| Parameter     | Description                                               |
+| ------------- | --------------------------------------------------------- |
+| Link          | a shared link to the file                                 |
+| submission_id | identifies the submission                                 |
+| attestation   | A boolean value indicating if the file is for attestation |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used to check status of upload|
+
+**Response:**
+
+| Key        | Description                                                                |
+| ---------- | -------------------------------------------------------------------------- |
+| dataset_id | identifies the data set the file will be associated with after validation. |
+| status     | Provides the current status of the upload			  |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
+| 400  | if the file type is invalid                                                                                            |
+
+#### POST submission/{submission_id}/upload/file
+
+An authenticated user can upload a file directly from their local machine to a data set in their submission.
+
+Asset_uuid can be included to retrieve a new access token if you were unable to complete the upload before the token
+expired. An error will be returned if the upload has already been completed. The user must own the submission and be
+the one who initiated the upload to retrieve an additional access token.
+
+If the size of the file being upload is <5GB, then a pre-signed URL will be returned.
```

Comment:
5GB is the maximum size fo the single file upload.   But AWS recommends that if you are using multipart uploads then use 100 MB.   "In general, when your object size reaches 100 MB, you should consider using multipart uploads instead of uploading the object in a single operation."  https://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu.html.

Reply:
I'll change this to be <100MB for a presigned url.